src/sspi/pku2u/cert_utils/validation.rs

```diff
@@ -0,0 +1,50 @@
+use picky_asn1_x509::signed_data::{CertificateChoices, SignedData};
+use picky_asn1_x509::{Certificate, PublicKey};
+use rsa::{BigUint, RsaPublicKey};
```

Comment:
Do you think we could use `picky` instead of depending on `rsa` directly? If I recall correctly, the logic in `picky::key::PublicKey` is roughly the same.
As a bonus, once `picky` gains support for elliptic curves, we get it in sspi-rs as well for free 🙂

Reply:
I just forgot that we have signature generation and verification implemented in picky. 
Done.